include/cuco/detail/static_map/static_map_ref.inl

```diff
@@ -196,16 +199,16 @@ class operator_impl<
   /**
    * @brief Inserts an element.
    *
-   * @tparam Value Input type which is implicitly convertible to 'value_type'
+   * @tparam ProbeKey Input key type which is convertible to 'key_type'
    *
    * @param group The Cooperative Group used to perform group insert
    * @param value The element to insert
    *
    * @return True if the given element is successfully inserted
    */
-  template <typename Value>
+  template <typename ProbeKey>
   __device__ bool insert(cooperative_groups::thread_block_tile<cg_size> const& group,
-                         Value const& value) noexcept
+                         cuco::pair<ProbeKey, mapped_type> const& value) noexcept
```

Comment:
I went with this version since it requires less changes and is more sane.

Reply:
why is `ProbeKey` being convertible to `key_type` while `payload` has to be `mapped_type`

Reply:
The "heterogeneous" part only affects the key, i.e., when introducing a custom `key_equal` functor or key `hash`. So we only need the tparam for the custom key type, while the payload can be the same as the map's `mapped_type`

Reply:
My point is that as long as the insert pair is convertible to the map's `value_type`, it should be fine. The payload can be any arbitrary types that can be convertible to `mapped_type`.

Reply:
Gotcha. Makes sense. Let me try out this change and see if it works.

Reply:
IIRC based on offline discussion we decided to leave it as is. Can you confirm?

Reply:
Addressed in [f5bf6d8](https://github.com/NVIDIA/cuCollections/pull/387/commits/f5bf6d839a313bfbec3be8820a0fc7daee4e8883)